tests/testthat/test-shinytest2-init.R

```diff
@@ -8,6 +8,65 @@ testthat::test_that("e2e: teal app initializes with no errors", {
   app$stop()
 })
 
+testthat::test_that("e2e: teal app initializes with sessionInfo modal", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = example_module(label = "Example Module")
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check if button exists.
+  button_selector <- "#teal-sessionInfo-button"
+  testthat::expect_equal(
+    app$get_text(button_selector),
+    "Session Info"
+  )
+
+  app$click(selector = button_selector)
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  # Check header and title content.
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-header > h4"),
+    "SessionInfo"
+  )
+  testthat::expect_equal(
+    app$get_text("#teal-sessionInfo-copy_button1"),
+    "Copy to Clipboard"
+  )
+  testthat::expect_equal(
+    app$get_text("#shiny-modal > div > div > div.modal-body > div > button:nth-child(2)"),
+    "Dismiss"
+  )
```

Comment:
This may be controversial, but how about checking the labels of buttons in a single swoop.
- There are 2 of each
- Have the correct label

If you don't like this, then just change to `button[data-dismiss]` instead of `nth-child(2)`

```suggestion
  # Check all buttons
  testthat::expect_setequal(
    testthat::expect_length(
      app$get_text(
        "#shiny-modal :where(#teal-sessionInfo-copy_button1, #teal-sessionInfo-copy_button2)"
      ),
      2
    ),
    "Copy to Clipboard"
  )
  
  testthat::expect_setequal(
    testthat::expect_length(
      app$get_text("#shiny-modal button[data-dismiss]"),
      2
    ),
    "Dismiss"
  )
```

Reply:
Or use `#shiny-modal [id^='teal-sessionInfo-copy_button']` instead of the `:where` pseudo class